File: python/Binary-Tree-Postorder-Traversal/Binary-Tree-Postorder-Traversal.py
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        if root == None:
            return res
        self.helper(root,res)
        return res
    def helper(self,root,res):
        if root is None:
            return
        self.helper(root.left,res)
        self.helper(root.right,res)
        res.append(root.val)

        
        
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        if root is None:
            return res
        s = []
        s.append(root)
        while s:
            node = s.pop()
            if node.left:
                s.append(node.left)
            if node.right:
                s.append(node.right)
            res.append(node.val)
        return res[::-1]

File: cpp/Binary-Tree-Postorder-Traversal/Binary-Tree-Postorder-Traversal.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> res;
        if (root==NULL)
            return res;
        helper(root,res);
        return res;
    }
    void helper(TreeNode* root, vector<int> &res)
    {
        if (root==NULL)
            return;
        helper(root->left,res);
        helper(root->right,res);
        res.push_back(root->val);
    }
};


class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> res;
        if (root==NULL)
            return res;
        stack<TreeNode*> s;
        s.push(root);
        while (!s.empty())
        {
            TreeNode* node = s.top();
            s.pop();
            if (node->left)
                s.push(node->left);
            if (node->right)
                s.push(node->right);
            res.push_back(node->val);
        }
        reverse(res.begin(),res.end());
        return res;
    }
};
